app/src/pages/BookmarksPage.vue

```diff
@@ -0,0 +1,59 @@
+<script lang="ts" setup>
+import ContentTile from "@/components/content/ContentTile.vue";
+import { appLanguageIdAsRef, userPreferencesAsRef } from "@/globalConfig";
+import {
+    db,
+    DocType,
+    useDexieLiveQueryWithDeps,
+    type ContentDto,
+    type Uuid,
+} from "luminary-shared";
+import { computed } from "vue";
+import { BookmarkIcon } from "@heroicons/vue/24/solid";
+
+// Extract bookmark keys
+const bookmarkKeys = computed(() => Object.keys(userPreferencesAsRef.value.bookmarks));
```

Comment:
you need to handle an empty userPreferences object (with no `bookmarks` key)
```suggestion
const bookmarkKeys = computed(() => Object.keys(userPreferencesAsRef.value.bookmarks || []));
```